Fix speech recognition example in documentation: button logic and error handling

STECHNO opened the pull request:
This pull request addresses two issues in the speech recognition example code:

1. Fixed the button rendering logic:
   Corrected the `recognizing` state in the conditional rendering of the Start/Stop button. The "Start" button now shows when not recognizing, and "Stop" when recognizing.

2. Improved error handling in the `handleStart` function:
   Added try/catch block and made the function asynchronous to properly handle potential errors during permission request and speech recognition start.

These changes improve the accuracy and robustness of the documentation example, ensuring that developers can implement speech recognition in their Expo projects more reliably.

Testing:
- The updated code has been tested in an Expo project to ensure it functions as expected.
- The Start/Stop button now correctly reflects the current state of speech recognition.
- Error handling has been verified to catch and log potential issues.

Please take a look at these changes and let me know if you need any more changes.

jamsch commented on README.md:
```diff
@@ -119,12 +121,14 @@ function App() {
         contextualStrings: ["Carlsen", "Nepomniachtchi", "Praggnanandhaa"],
       });
-    });
+    } catch (error) {
+      console.error("Error starting speech recognition:", error);
+    }
```
This hopefully shouldn't be required as the native iOS and Android implementations for `.start()` handle their own errors and emit those errors to the `"error"` event. You can listen to that with `useSpeechRecognitionEvent("error", ...)`. 

Likewise, with the `requestPermissionsAsync` function, it would only throw if the project was using bare React Native without `expo-modules-core` (which is present in every Expo project).

Let me know though if you did catch any uncaught errors as I don't intended for these functions to ever throw.

jamsch commented on README.md:
```diff
@@ -103,14 +103,16 @@ function App() {
       }
       // Start speech recognition
-      ExpoSpeechRecognitionModule.start({
+      await ExpoSpeechRecognitionModule.start({
```
This function is synchronous so it shouldn't need to be awaited. Asynchronous behavior for this function is captured through the events (start,stop,error,speechstart,result etc)

jamsch reviewed (changes requested):
Hey @STECHNO, good spotting!

Just a few small changes necessary

STECHNO commented:
Thank you @jamsch for the feedback. I've made the following changes:

1. Removed the try-catch block from the handleVoice function.
2. Adjusted the error handling to use useSpeechRecognitionEvent("error", ...) instead.
3. Removed the async/await from ExpoSpeechRecognitionModule.start() as it's synchronous.

Regarding your question about uncaught errors: No, I haven't encountered any uncaught errors in my testing. I initially added the try-catch block thinking it might help with potential permission-related errors on older Android versions. However, I agree with your assessment that it's unnecessary, especially given that Expo handles these scenarios. I've removed that code as per your suggestion.

I've tested these changes, and everything seems to be working.

Thank you again for your guidance and for helping improve the documentation!

jamsch reviewed (approved):
Thank you!
